ipatests/test_integration/test_trust.py

```diff
@@ -635,7 +635,8 @@ def test_server_option_with_unreachable_ad(self):
             # This checks that our setup is correct
             result = self.master.run_command(
                 ['ipa', 'trust-add', self.ad.domain.name,
-                 '--admin', 'Administrator', '--password'], raiseonerr=False)
+                 '--admin', 'Administrator', '--password'], raiseonerr=False,
+                stdin_text=self.master.config.ad_admin_password)
```

Comment:
Is this required only on ipa-4-6 branch? I could not spot the code that would explain this difference.

Reply:
1. I admit this is a mistake in tests.
2. In master this mistake does not prevent test from passing: http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/7aafc9e4-68be-11ea-97e2-fa163e3ad6c5/report.html
```
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd544:transport.py:513 RUN ['ipa', 'trust-add', 'ad.test', '--admin', 'Administrator', '--password']
DEBUG    ipatests.pytest_ipa.integration.host.Host.master.cmd544:transport.py:558 ipa: ERROR: CIFS server communication error: code "3221225653", message "{Device Timeout} The specified I/O operation on %hs was not completed before the time-out period expired." (both may be "None")
```

In master branch empty password does produce error message when AD controller is available. So it looks as if the checks for password and connectivity are made in different order. Unfortunately I do not know if it is a regression or intended change (and not quite sure if that is the real root cause).

I will add a separate PR for master, ipa-4-7 and ipa-4-8 for the fix but I am not sure how to deal with this behavior change.

Reply:
Created #4399